pkg/transforms/argoapplication.go

```diff
@@ -122,6 +129,16 @@ func ArgoApplicationResourceBuilder(a *ArgoApplication) *ArgoApplicationResource
 		}
 	}
 
+	// if there is operationState.message, append it to the error condition property
+	if a.Status.Health.Status != "Healthy" || a.Status.Sync.Status != "Synced" {
+		if a.Status.OperationState != nil && a.Status.OperationState.Message != "" {
+			truncatedMessage := TruncateText(a.Status.OperationState.Message, 512)
+			if truncatedMessage > "" {
+				node.Properties["_internalCondition"+"OperationError"] = truncatedMessage
```

Comment:
Maybe the name of this property should be `_operationStateMessage`?
The `_` means that it's an internal property, so you can omit the word "internal".  
Could you also remove "internal" from the condition above?

Reply:
@jlpadilla  Here is what I updated
- removed the `internal` keyword. So the condition property naming convention is `_condition<type>`  

- The <type> is from argocd status condition type, its naming convention is <error type> with the suffix Error or Warning
https://github.com/argoproj/argo-cd/blob/22ea51d4e40f4a57d825eeb8c0f483bc7150d340/pkg/apis/application/v1alpha1/types.go#L963-L980

- The error type for the operationState.message is called `_conditionOperationError` to keep the consistency with other Argocd app condition types.